h2/src/main/org/h2/expression/Function.java

```diff
@@ -1493,7 +1497,46 @@ private Value getValueWithArgs(Session session, Expression[] args) {
             break;
         case EXTRACT: {
             int field = getDatePart(v0.getString());
-            result = ValueInt.get(DateTimeUtils.getDatePart(v1, field));
+            // Normal case when we don't retrieve the EPOCH time
+            if(field != EPOCH) {
+                result = ValueInt.get(DateTimeUtils.getDatePart(v1, field));	
+            }else {
+            	// Case where we retrieve the EPOCH time.
+            	// First we retrieve the dateValue and his time in nanoseconds.
+            	long[] a = DateTimeUtils.dateAndTimeFromValue(v1);
+                long dateValue = a[0];
+                long timeNanos = a[1];
+                // We compute the time in nanoseconds and the total number of days.
+        		BigDecimal timeNanosBigDecimal = new BigDecimal(timeNanos);
+        		BigDecimal numberOfDays = new BigDecimal(DateTimeUtils.absoluteDayFromDateValue(dateValue));
+        		BigDecimal nanosSeconds = new BigDecimal(1_000_000_000);
+        		BigDecimal seconds = new BigDecimal(60);
+        		BigDecimal secondsPerDay = new BigDecimal(DateTimeUtils.SECONDS_PER_DAY);
+                // Case where the value is of type time e.g. '10:00:00'
+            	if(v1 instanceof ValueTime) {
+            		// In order to retrieve the EPOCH time we only have to convert the time 
+            		// in nanoseconds (previously retrieved) in seconds.
+            		result = ValueDecimal.get(timeNanosBigDecimal.divide(nanosSeconds));
+            	} else if (v1 instanceof ValueDate) {
+            		// Case where the value is of type date '2000:01:01', we have to retrieve the total 
+            		// number of days and multiply it by the number of seconds in a day.
+            		result = ValueDecimal.get(numberOfDays.multiply(secondsPerDay));
+            	}else if(v1 instanceof ValueTimestampTimeZone) {
+            		// Case where the value is a of type ValueTimestampTimeZone ('2000:01:01 10:00:00+05).
+            		// We retrieve the time zone offset in minute
+            		ValueTimestampTimeZone v = (ValueTimestampTimeZone) v1;
+            		BigDecimal timeZoneOffsetMins = new BigDecimal(v.getTimeZoneOffsetMins());
```

Comment:
`BigDecimal timeZoneOffsetSeconds = new BigDecimal(v.getTimeZoneOffsetMins() * 60);`

And you don't need seconds constant above.